server/resp/src/main/java/org/infinispan/server/resp/commands/json/JSONOBJKEYS.java

```diff
@@ -51,9 +50,11 @@ static BiConsumer<List<List<byte[]>>, ResponseWriter> newIntegerOrErrorWriter(by
       return (c, writer) -> {
          if (c == null || c.size() == 0) {
             writer.nulls();
+            return;
```

Comment:
Make sure this stays in though, as it looks like a bug that we must not be testing properly.